doc/source/cluster/kubernetes/user-guides/upgrade-guide.md

```diff
@@ -26,54 +26,23 @@ If you want to understand the reasoning behind the CRD versioning plan, see [ray
-  * Example 1: Upgrade KubeRay from v0.6.0 to v1.0.0 without deleting the old CRD.
-    ```shell
-    # Install KubeRay v0.6.0 and CRD v1alpha1
-    helm install kuberay-operator kuberay/kuberay-operator --version 0.6.0
-
-    # The following instruction uninstalls only KubeRay v0.6.0. It does not uninstall CRD v1alpha1.
-    helm uninstall kuberay-operator
-
-    # Install KubeRay v1.0.0. Because the CRD already exists, the Helm instruction does not install the new CRD.
-    helm install kuberay-operator kuberay/kuberay-operator --version 1.0.0
-
-    # Check CRD
-    kubectl describe crd rayclusters.ray.io | grep v1
-    # You can only see "Name: v1alpha1", and cannot see "Name: v1".
-
-    # Install RayCluster v1.0.0 which uses CRD v1.
-    helm install raycluster kuberay/ray-cluster --version 1.0.0
-    # Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest:
-    # resource mapping not found for name: "raycluster-kuberay" namespace: "" from "": no
-    # matches for kind "RayCluster" in version "ray.io/v1"
-    # ensure CRDs are installed first
-    ```
-  * Example 2: Upgrade KubeRay from v0.6.0 to v1.0.0 with deleting the old CRD.
-    ```shell
-    # Install KubeRay v0.6.0 and CRD v1alpha1
-    helm install kuberay-operator kuberay/kuberay-operator --version 0.6.0
-
-    # The following instruction uninstalls only KubeRay v0.6.0. It does not uninstall CRD v1alpha1.
-    helm uninstall kuberay-operator
-
-    # Delete CRDs. Note that deleting the CRD causes a cascading deletion of custom resources.
-    kubectl delete crd rayclusters.ray.io
-    kubectl delete crd rayjobs.ray.io
-    kubectl delete crd rayservices.ray.io
-
-    # Install KubeRay v1.0.0 and new CRD including v1.
-    helm install kuberay-operator kuberay/kuberay-operator --version 1.0.0
-
-    # Check CRD
-    kubectl describe crd rayclusters.ray.io | grep v1
-    # You can see both "Name: v1alpha1" and "Name: v1".
-
-    # Install RayCluster v1.0.0 which uses CRD v1.
-    helm install raycluster kuberay/ray-cluster --version 1.0.0
-    ```
+Due to reliability and security implication of webhooks, KubeRay doesn't support a conversion webhook to convert v1alpha1 to v1 APIs.
+
+To upgrade the KubeRay version, it's highly recommended to follow these steps in order:
+1. Upgrade the CRD manifest, containing new fields added to the v1 CRDs.
+2. Upgrade the kuberay-operator image to the new version
+3. Verify successful upgrade
+
+Here's an example when upgrading KubeRay from v1.0.0 to v1.1.0:
+```
+# upgrade the CRD to v1.1.0
+# NOTE: here we use kubectl because Helm does not support lifecycle management of CRDs
+# See the Helm documentation for more details: https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#some-caveats-and-explanations
+$ kubectl apply --server-side --force-conflicts -k "github.com/ray-project/kuberay/ray-operator/config/crd?ref=v1.1.0"
```

Comment:
The `--force-conflict` is needed here to resolve server-side conflicts with helm:
```
$ kubectl apply --server-side -k "github.com/ray-project/kuberay/ray-operator/config/crd?ref=v1.1.0"
Apply failed with 2 conflicts: conflicts with "helm" using apiextensions.k8s.io/v1:
- .metadata.annotations.controller-gen.kubebuilder.io/version
- .spec.versions
Please review the fields above--they currently have other managers. Here
are the ways you can resolve this warning:
* If you intend to manage all of these fields, please re-run the apply
  command with the `--force-conflicts` flag.
* If you do not intend to manage all of the fields, please edit your
  manifest to remove references to the fields that should keep their
  current managers.
* You may co-own fields by updating your manifest to match the existing
  value; in this case, you'll become the manager if the other manager(s)
  stop managing the field (remove it from their configuration).
See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts
```

I think it's fine to force resolve these conflicts because Helm is not managing CRDs after initial install anyways.